pkg/virt-controller/watch/migration.go

```diff
@@ -389,11 +399,39 @@ func (c *MigrationController) createTargetPod(migration *virtv1.VirtualMachineIn
 func (c *MigrationController) sync(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, pods []*k8sv1.Pod) error {
 
 	var pod *k8sv1.Pod = nil
+	conditionManager := controller.NewVirtualMachineInstanceMigrationConditionManager()
 	podExists := len(pods) > 0
 	if podExists {
 		pod = pods[0]
 	}
 
+	if vmi != nil && conditionManager.HasCondition(migration, virtv1.VirtualMachineInstanceMigrationAbortRequested) {
```

Comment:
```vmi != nil && vmi.DeletionTimestamp != nil && !vmi.IsFinal()```  instead of checking the condition here?

Reply:
I don't mind. Not sure why did we need a condition in the first place.

Reply:
The condition is not there for the controller itself. The condition is there for other controllers, the ui and users.

Reply:
i suppose this also has an implication for updates/rollbacks.

the controller loop shouldn't be reacting to the Status section of our API during sync because it's possible the control loop isn't aware of the condition after a downgrade of the virt-handler component.